packages/api/tests/app/test-middleware.js

```diff
@@ -0,0 +1,43 @@
+const test = require('ava');
+const express = require('express');
+const boom = require('express-boom');
+const request = require('supertest');
+
+const { EcsStartTaskError } = require('@cumulus/errors');
+
+const { asyncOperationEndpointErrorHandler } = require('../../app/middleware');
+
+const app = express();
+app.use(boom());
+
+// Set up fake endpoint to just test middleware
+app.post(
+  '/fake-async-endpoint',
+  () => {
+    throw new Error('failed to start');
+  },
+  asyncOperationEndpointErrorHandler
+);
+
+app.post(
+  '/fake-async-endpoint2',
+  () => {
+    throw new EcsStartTaskError('failed to start');
+  },
+  asyncOperationEndpointErrorHandler
+);
+
+test('request to replays endpoint returns 500 if starting ECS task throws unexpected error', async (t) => {
+  const response = await request(app)
+    .post('/fake-async-endpoint')
+    .expect(500);
+  t.is(response.status, 500);
```

Comment:
If this is anything other than `500`, the `.expect(500)` line above is going to cause an exception to be thrown, and this line will never be reached. I'd probably drop the `.expect(500)`. Same thing for the other test in this file.